pkg/scheduler/framework/frameworkutils.go

```diff
@@ -135,30 +139,21 @@ func crossReferencePickedCustersAndObsoleteBindings(
 
 		// The binding's target cluster is picked again in the current run; yet the binding
 		// is originally created/updated in accordance with an out-of-date scheduling policy.
+		// Add the binding to the toPatch list.
+		toPatch = append(toPatch, patchBinding(binding, scored, policy))
+	}
 
-		// Update the binding so that it is associated with the latest score.
-		updated := binding.DeepCopy()
-		affinityScore := int32(scored.Score.AffinityScore)
-		topologySpreadScore := int32(scored.Score.TopologySpreadScore)
-		updated.Spec.ClusterDecision = placementv1beta1.ClusterDecision{
-			ClusterName: scored.Cluster.Name,
-			Selected:    true,
-			ClusterScore: &placementv1beta1.ClusterScore{
-				AffinityScore:       &affinityScore,
-				TopologySpreadScore: &topologySpreadScore,
-			},
-			Reason: pickedByPolicyReason,
+	for _, binding := range unscheduled {
```

Comment:
For this one, originally we agreed that unscheduled is a terminal state for bindings; we do not bring it back to life as it risks contention.

Reply:
One of the tricky points is that, a binding can reach the unscheduled states from two other states: scheduled, and bound. If we revive an unscheduled binding, we cannot reliably tell which state to revive it back to at this point.

Reply:
It may involve some minor side effects, mostly related to rolling updates (if applicable).